doc/architecture.md

```diff
@@ -0,0 +1,136 @@
+Architecture
+============
+
+This section describes how a Management Pack works, and how it interacts with the VMware 
+Aria Operations Collector.
+
+## Management Pack
+A management pack consists of content, metadata, and an adapter.
+
+* Content includes Dashboards, Reports, Icons, Symptoms, Alerts, Traversals, others.
+* Metadata includes the describe.xml file that describes an adapter, and the 
+  `manifest.txt` file that describes the management pack as a whole.
+* The Adapter is the code that performs the collection and sends the data to VMware Aria 
+  Operations.
+
+### Adapter Container Image
+
+The adapter container image consists of two parts:
+* A REST server specified by [Collector Framework 2](../vmware_aria_operations_integration_sdk/api/vmware-aria-operations-collector-fwk2.json).
+* The adapter code that performs the collection.
+
+In an adapter created using the SDK, the REST server is provided inside a base container
```

Comment:
This section starts to get pretty wordy and hard to follow. Maybe we could break it up a bit. For example, show an example snippet of a Dockerfile. Just in general I think this section is chaining together too many MP/Aria Ops-specific concepts that are new to the reader, making it difficult to follow.

Reply:
I added an example here, but otherwise I didn't change this section too much. I agree this is somewhat dense, but I don't know if I can avoid that in general for what I want this document to accomplish.